src/include/OpenImageIO/strutil.h

```diff
@@ -79,16 +80,54 @@ OIIO_NAMESPACE_BEGIN
 /// @brief     String-related utilities.
 namespace Strutil {
 
+/// Output the string to the FILE* stream in a synchronized fashion, so that
+/// buffers are flushed and internal mutex is used to prevent threads from
+/// clobbering each other -- output strings coming from concurrent threads
+/// may be interleaved, but each string is "atomic" and will never splice
+/// each other character-by-character.
+void OIIO_API sync_output (FILE *file, string_view str);
+
+
 /// Construct a std::string in a printf-like fashion.  In other words,
 /// something like:
 ///    std::string s = Strutil::format ("blah %d %g", (int)foo, (float)bar);
 ///
-/// The printf argument list is fully typesafe via tinyformat; format
-/// conceptually has the signature
-///
-/// std::string Strutil::format (const char *fmt, ...);
-TINYFORMAT_WRAP_FORMAT (std::string, format, /**/,
-    std::ostringstream msg;, msg, return msg.str();)
+/// Uses the tinyformat library underneath, so it's fully type-safe, and
+/// works with any types that understand stream output via '<<'.
+template<typename T1, typename... Args>
+std::string format (string_view fmt, const T1& v1, const Args&... args)
```

Comment:
Turns out the base case is needed after all, because the underlying Tinyformat template does not provide one. Their rationale is that you don't need the formatting if there are no arguments, but I kind of like the symmetry, and since our printf/fprintf also ensure that the contents are output atomically in a threaded environment, I think it's nice to be able to use them without arguments.